### PR TITLE
Cere-station [100% free lag edition]

### PR DESCRIPTION
**Pull request by S34NW**
## What Does This PR Do
Adds Cere station, a currently para-unique map. It's an older map that used to be on TG (is not anymore), but modernised for para.
This is attempt 2 at getting this in, but feating extreme further optimisation and design choices that leads to playable CPU load.

Cere is not "bigger" than existing maps, namely delta, but is more spread out. Walking can take a while here, best hope the quantum pads work.

Cere is radically different from our existing station designs, with a heavy focus on being spread out/isolated, a less "high tech" vibe with more of a feeling that NT did the bare minimum of digging to get this functional as a station.

Cere has:

- Looping exterior space disposals
- Prisoner delivery chute to cut down on walking for security officers
- Roundstart quantum pads for ease of travel between the four map corners
- 7 (8 counting AI sat) asteroids housing various departments, with remote lockdown capabilities from the bridge or head offices.
- Distributed powergrid, with "subnets" and "substations" for each department/asteroid to increase powersink resiliency
- No gateway
- DIY cult bases (or hangout areas) using rock walls to dig out your own whatever
- AI upload on the science asteroid
- Ancient rock, a new rock type for cere that can only be mined with a pickaxe or its subtypes (or explosions)
- Cold ancient rock, like regular ancient rock but can only be mined with a diamond pick (Lines the station perimiter to stop people mass-spacing the asteroid)
- False walls for ancient rock, used by putting sand you can mine from the ancient rock or the sand on a displaced girder

Todo:

- [x] Consider cycling airlocks for all bridge-side access
- [x] Squash bugs
- [ ] Improve UX

## Why It's Good For The Game
More unique maps that aren't just box but rotated 90 degrees.

## Testing
Multiple TMs

## Changelog
:cl:
add: Added cere station, a new map
add: Ancient rock, a new rock type for cere that can only be mined with a pickaxe or its subtypes
add: Cold ancient rock, like regular but can only be mined with a diamond pick
add: False walls for ancient rock, used by putting sand on a displaced girder
/:cl:

### CODE REVIEW

**Comment by S34NW** [comment]
Oops i left in TM only changes, reverting those

**Comment by Carthusia** [comment]
I do like the maps layout, but during the last TM, like 90% of the stations rock was mined away.

My prime concern is this will happen every round

**Comment by PacifistDalek** [comment]
YESSSS IT LIIIIVEEESSS

**Comment by AffectedArc07** [comment]
Well this puts them through their paces 
image

**Comment by TrashDoxx** [comment]
image

**Comment by TrashDoxx** [comment]
Where's that foxy guy?

**Comment by Komrad822** [comment]
Isnt important nor do I see it happening since you'd have to move a lot around for this but airlock system for the SM and possible atmos access airlock from the side?

**Comment by Pennwick** [comment]
Whats the power draw on the station compared to how much solars can output? A lot of the arrays seem smaller and covered from the various directions. With Substations you could possibly down the draw from certain areas but with these sunblocked solars I doubt they'd produce enough for even one substation to run.

**Comment by S34NW** [comment]
> 



> Whats the power draw on the station compared to how much solars can output? A lot of the arrays seem smaller and covered from the various directions. With Substations you could possibly down the draw from certain areas but with these sunblocked solars I doubt they'd produce enough for even one substation to run.

The solar issue is quite relevant, I likely need to buff them as currently it's a bit all over the place.

**Comment by Qwertytoforty** [comment]
no ion rifle in armory or ablative

**Comment by Samman166** [comment]
Issue list

-No disk compartmentalizer

-No Sterile masks for medbay

-Issues reaching cycle airlock buttons

-Medbay fridge access can't be opened with just medbay access

-Few maints rooms with vents for terrors/blob/xenos

**Comment by Mooballoono** [comment]
i encountered 3 issues in the bar
1. the disposals behind the bar counter has 2 disposals and one of them shoots stuff out of it
2. the bar maints access is a hydroponics door for no reason
3. in the maints behind the hydroponics door there is a maints door above that goes into an airlock

**Comment by Qwertytoforty** [comment]
copypaste from fax

From: The Desk of Nanotrasen Representative Cassidy Berkhillmer


Command


This section is dedicated to critiques provided to me by command staff. Head of Personnel Trystan Bellfred notes that at the beginning of the shift, an ID Compartmentalizer was not present (this has since been remedied). In addition, the HoP line's ticket counter/dispenser is at the end of the line, rather than at the beginning.


Captain Joseph Snow has shown a particular concern for the lack of easy transport to the medical bay compared to the rest of the station, given its massive importance to a smooth-running workspace. The department lockdown buttons seem like may come in quite handy, though he has yet to employ them.


Cargo


Quartermaster Amanda Ripley notes that Cargo has been constructed very well. The amount of provided space is excellent, and she relishes the easy access to disposals.


Engineering


Atmospheric Technician Corvidium has expressed a concern for the lack of EVA access afforded to Atmospherics personnel. Apparently they aren't able to properly use any of the EVA airlocks, including the ones inside of their own checkpoints. In addition, he notes that a lack of access to the various engineering vending machines make conducting repairs a lot harder.


On a more positive note, I've been told that the engineering and atmospherics checkpoints are quite useful, given the size of the NSS Farragus. I recommend giving a commendation to whoever had the foresight to include them.


Medbay


Chief Medical Officer Maco Manz has been quite vocal about an issue with our disposal units. Something is happening that causes trash to be delivered to his office and the medical bay entrance. Other than that, he seems to be satisfied with the design of Medical.


Science


Research Director Roland Firsky notes that the server room as it is currently constructed makes it so you have to be exposed to dangerously cold air in order to access the server console. Outside of that, he expresses praise for the department design.


Security


Magistrate FIREWALL has reported a potential security risk in the NSS Farragus's permabrig. Apparently there is a very large chunk of rock that used to lead straight into maintenance (it has since been reinforced). Since pickaxes are provided to the prisoners, I think it would be wise to send a specialist to properly reinforce perma at a later date.


Head of Security Moi's only complaint appears to be a lack of a paperbin in her office. Aside from that, she finds the brig and its systems satisfactory.


General Station Status


On a more general note, I've noticed that crew utilities such as cryogenic storage, botany, and the very convenient asteroid telepads are somewhat tucked away, making them quite difficult to spot. They may require additional signage in the future.

**Comment by S34NW** [comment]
Thanks all for feedback thus far, most of these issues should now be resolved. Keep the feedback coming!

**Comment by Sollessa** [comment]
I don't know exactly what the issue is but the station seems to have massive power issues.  Any delays at all in getting the engine running seems to cause massive black outs.  I'm not entirely sure how the station is wired but I suspect that the issue is that there are just too many SMES drawing power from the engine because all of the auxiliary SMES start on with maximum draw.  If that's not the issue then another thought is to increase the number of rad collectors on the Supermatter.

From what it seems there are like 7 SMES that are fed by the engine, all at maximum draw.  That's way too many for the base SM setup and so something has to be done.  I understand that setting up power shouldn't be brainless like it used to be but it also shouldn't take several engineers 30+ minutes to figure out as they try to decipher the far more complicated wiring setup for the station.

**Comment by Adrer** [comment]
image
This room seems terribly dangerous. Someone could VERY easily walk into this room, get rolled by the conveyor belts into an open disposals chute. Probably good to put some plastic flaps in front of.

**Comment by CinnamonSnowball** [comment]
Okay, after a full medical shift on Cere, the Medbay is.. a bit of a mess, and unfortunately feels like fixing some of the bigger issues would require a somewhat big redesign; in general, some less important parts of it are very large for no particular reason, while the more important areas are fairly small and/or stuffy.

Layout:
- it's awkward to have the main Medbay room simultaneously be the waiting room, the examination room, and the showers room; patients and docs treating them block the way fairly often
- the Medbay-facing chemistry fridge - where docs restock their meds - is in a bit of an awkward place, sitting right in front of the entrance airlock; I'd say it could switch places with the Chemistry airlock for better flow, but that one is *also* right in front of the entrance to Staff Room/Medical Supplies corridor, so it's not great one way nor the other
- Chemistry itself is very stuffy and could use to be slightly bigger, maybe redesigned a bit as well to 
- the cloning lab is massive, and the relevant machinery is mostly concentrated in two corners of the room with a lot of empty space in between; notably, the Cryocell is across the entire room from the cloning pod
- there's no dedicated Cryogenics room - the single cloning lab Cryocell is all the Medbay has, and it's a bit out of the way
- three surgery rooms roundstart is a bit extra, I feel like the two tiny ones should be expanded cannibalizing on the larger OR
- despite three ORs, there's no surgery storage, and the surgery rooms also don't have a doctor equipment locker for surgeons which means dragging duffelbags/defibs/IV drips for OR setup from all the way out in Supplies
- although I'm not an expert on room standards, the placement of scrubbers and vents all over Medbay seems a bit dubious

Smaller issues:
- the exam room (aka main Medbay) is missing some tables here and there; there's not even a place to even put a Defib down besides the floor
- the blood crates in main Medbay room and Medical Supplies don't seem to be the regular variant, instead coming with extra oxygen tanks and IV drips which get sucked into crates whenever someone opens and closes them (worse yet, the IV drips pick up one of the blood bags when the map is initialized, so you have to take it off before using the IV drip)
- there are only 5 NanoMed Plus vendors the doctors have access to, which is one less than Cyberiad/Delta; maybe the Virology vendor could be moved somewhere into Medbay, since Virologists don't need their own NanoMed anyway?
- 5 Sleepers roundstart might be a bit much?
- are there any boxes of white latex gloves anywhere? couldn't find one in about five minutes of looking (on other maps they're often in reception, exam room, supplies room, or surgery storage, for example)
- similarly, the department looks a bit low on roller beds, only ones I can see are near the tiny patient room
- Medical Doctors for some reason have access to Genetics Research
- there's a button on the wall next to the Genetics Research airlock named "Cloning Door" that seems to do nothing?
- the cloning lab air alarm doesn't do its job very well since it sits in the little cryo gas canister closet
- the window tint switches in the ORs sit on the same tile as a light tube, which looks a little weird; IMO the window tint switches should also switch places with the light switches (i.e. have the holosign and window tint switches together in reach of an operating surgeon, and the light switch near the entrance)
- the OR airlocks could be changed to the electrochromic window variants, currently they're solid despite having windows right next to them
- I think the long stretch of garden in main Medbay hallway could be split in two to make it easier to get from the surgery rooms to the closest NanoMed in case supplies are suddenly needed in surgery
- nitpick, but the lockers in Medical Supplies are weirdly shuffled, maybe they could be sorted a bit more sensibly? perhaps the doctor equipment locker that sits in the corridor outside could be sucked in as well, as it feels a bit out of place there
- the table in the corner of medical storage with medHUDs/space cleaner/soap is blocked by the blood crate and doctor gear locker next to it
- the two exits out of Paramed's office (the shutters and the airlock) awkwardly sit on the opposite sides of the atmos-restricting hallway airlock triplet; also, while giving the Paramedic a bed is a nice touch, putting it next to the airlock is mildly cursed room design

**Comment by Alantodne** [comment]
As a engineer I noticed that the railings around the disposal lines seem inconsistent if they run east/west vs the ones that run north/south. The ones that run east/west can be walked down but not up. You have to hop the railing to go north on the east/west running ones. while for the north and south ones, you cant cross. I got the error "cant cross because a railing is in the way" or something similar. I think the east and west ones are two lines of rails that face down, while the ones that run north and south are outer facing. Might be better to rotate it so that if someone spaces themselves via disposals, they dont accidentally get on the other side of the rail and die(its already a death trap, might be better to make it SLIGHTLY less deadly)

**Comment by Miraviel** [comment]
Medbay review/suggestions

(1) If possible, put two surgery cleaners here instead of one, one for each OR (we usually put them in the surgical duffel so we can keep the place clean)

image

(2) Surgeries have no IV drips

image

(3) Please remove this directional window, it blocks move flow when we move diagonally

image

image

(4) If possible, move the wall-mounted fuel tanks next to chem dispensers, it's just so much more comfortable to not having to walk around

image

(5) This freezer crate is horribly blocking movement, resulting in two, one-tile long corridor

image

(6) Break room is way too big imo, if you could decrease its height by 1, the storage (which is more important function-wise) could be more spacious. maybe

image

(7) No clue why there is a destination tagger and such here?

image

(8) Water machine, missing box of paper cups

image

(9) Purely QoL, but if you could make the tile below the showers high-traction tiles, I'd send you a kiss

image

**(10) Sean where is the medical wrench???????**

image

(11) Another QoL  request: replace this table with a medical records console so diligent coroners can update status records (when they cannot do so by examining bodies):

image

**Comment by Miraviel** [comment]
(12) Morgue lacks disposals, cannot use the shipping label feature without one (sending back ID and PDA to HOP)

**Comment by Miraviel** [comment]
(13) This chem windoor (north one) cannot be opened by doctor access

image

**Comment by CinnamonSnowball** [comment]
> (8) Water machine, missing box of paper cups

To be frank, the water coolers already have like 25 built-in paper cups that virtually never get used up, don't think an extra box is all that necessary

**Comment by Miraviel** [comment]
> > (8) Water machine, missing box of paper cups
> 
> To be frank, the water coolers already have like 25 built-in paper cups that virtually never get used up, don't think an extra box is all that necessary

I forgor, you are right

**Comment by S34NW** [comment]
> As a engineer I noticed that the railings around the disposal lines seem inconsistent if they run east/west vs the ones that run north/south. The ones that run east/west can be walked down but not up. You have to hop the railing to go north on the east/west running ones. while for the north and south ones, you cant cross. I got the error "cant cross because a railing is in the way" or something similar. I think the east and west ones are two lines of rails that face down, while the ones that run north and south are outer facing. Might be better to rotate it so that if someone spaces themselves via disposals, they dont accidentally get on the other side of the rail and die(its already a death trap, might be better to make it SLIGHTLY less deadly)

Thanks, resolved

**Comment by S34NW** [comment]
> Medbay review/suggestions

All points bar 1 and 8 addressed

**Comment by Sollessa** [comment]
This one is silly but you can't access the bookshelf in the NT Rep's room without climbing into the trash or something.  Just swap the position of the NT flag and the filing cabinet.
NTR

**Comment by Sollessa** [comment]
Station needs more flashlights in Engineering and other Engineering spaces.  Can't find a single one.

**Comment by Sollessa** [comment]
Sorry to be saying so much but I'm playing my first round as AI and there is a relatively criminal lack of holopads.  Some departments have plenty but then security has zero?  I also don't see any in any of the hallways.  I know it seems like a relatively minor thing but holopads are SUPER useful for the AI as a way to communicate easily, especially in times of comms outages.

**Comment by Komrad822** [comment]
>Station needs more flashlights in Engineering and other Engineering spaces. Can't find a single one.

Check fire lockers, they have lights in them.

**Comment by Sollessa** [comment]
> > Station needs more flashlights in Engineering and other Engineering spaces. Can't find a single one.
> 
> Check fire lockers, they have lights in them.

That's fine and all but it'd be nice if there was just a few sitting on the table in engineering like there is on other stations.  I know that if I was to scour the map I could find some but that feels unintuitive when you're playing the one department that regularly spends time in dark places and they don't have something like flashlights around.  I don't wear a hardsuit everywhere and it feels a bit silly to have to go on a scavenger hunt for a flashlight.  I don't think there's any fire closets in Engineering.

**Comment by efzapa** [comment]
whu

apperently this one tile below gene gets irradiated for some reason despite being in maints

**Comment by Sollessa** [comment]
Oh, on the note of maintenance and rad storms, it seems like a lot of the maintenance areas around arrivals and medical are unshielded.  I was scooting around as a borg in the maintenance areas adjacent to the gravity gen and the entire area was flooded with radiation.  Unsure if this is intentional or not.

**Comment by S34NW** [comment]
> whu
> 
> apperently this one tile below gene gets irradiated for some reason despite being in maints

This is required. The tiles that maint APCs are on are technically part of the room area, not maint area.

**Comment by S34NW** [comment]
> > > Station needs more flashlights in Engineering and other Engineering spaces. Can't find a single one.
> > 
> > 
> > Check fire lockers, they have lights in them.
> 
> That's fine and all but it'd be nice if there was just a few sitting on the table in engineering like there is on other stations. I know that if I was to scour the map I could find some but that feels unintuitive when you're playing the one department that regularly spends time in dark places and they don't have something like flashlights around. I don't wear a hardsuit everywhere and it feels a bit silly to have to go on a scavenger hunt for a flashlight. I don't think there's any fire closets in Engineering.

Added

**Comment by S34NW** [comment]
> Oh, on the note of maintenance and rad storms, it seems like a lot of the maintenance areas around arrivals and medical are unshielded. I was scooting around as a borg in the maintenance areas adjacent to the gravity gen and the entire area was flooded with radiation. Unsure if this is intentional or not.

These airlock access areas are part of the hallway, technically speaking. Making them require their own APC just for one small room seems like poor form to me. Perhaps this can be revisited at a later date

File: code/game/objects/structures/girders.dm [hunk 1]
```diff
@@ -117,6 +117,37 @@
...
+					to_chat(user, "<span class='notice'>You create a false wall. Push on it to open or close the passage.</span>")
+					var/obj/structure/falsewall/rock_ancient/FW = new (loc)
+					transfer_fingerprints_to(FW)
+					qdel(src)
+			else
+				if(A.get_amount() < 2)
+					to_chat(user, "<span class='warning'>You need at least two [A] to add plating!</span>")
+					return
+				to_chat(user, "<span class='notice'>You start adding [A]...</span>")
+				if(do_after(user,40, target = src))
```
Comment (hal9000PR) [inline code/game/objects/structures/girders.dm:139]:
```suggestion
				if(do_after(user, 4 SECONDS, target = src))
```

File: code/game/objects/structures/girders.dm [hunk 2]
```diff
@@ -117,6 +117,37 @@
...
+					if(!src || !A || A.get_amount() < 2)
+						return
+					A.use(2)
+					to_chat(user, "<span class='notice'>You add [A].</span>")
+					var/turf/Tsrc = get_turf(src)
+					Tsrc.ChangeTurf(/turf/simulated/mineral/ancient)
```
Comment (hal9000PR) [inline code/game/objects/structures/girders.dm:145]: this is the worst var name ever

File: code/game/objects/structures/girders.dm [hunk 3]
```diff
@@ -117,6 +117,37 @@
 					qdel(src)
 				return
 
+		if(istype(W, /obj/item/stack/ore/glass/basalt))
+			var/obj/item/stack/ore/glass/basalt/A = W
+			if(state == GIRDER_DISPLACED)
+				if(A.get_amount() < 2)
+					to_chat(user, "<span class='warning'>You need at least two [A] to create a false wall!</span>")
+					return
+				if(do_after(user, 20, target = src))
```
Comment (hal9000PR) [inline code/game/objects/structures/girders.dm:126]:
```suggestion
				if(do_after(user, 2 SECONDS, target = src))
```

File: code/game/objects/structures/girders.dm (hunk 2 above)
Comment (S34NW) [reply code/game/objects/structures/girders.dm:145]: I literally just copy-pasted girder code above, this whole proc is a mess

**Review by hal9000PR — APPROVED** [review]
fuck it we ball

**Review by Sirryan2002 — APPROVED** [review]
Lets go Cereal station

**Pull request merged**